libcfnet/tls_generic.c

```diff
@@ -818,17 +819,21 @@ void TLSSetDefaultOptions(SSL_CTX *ssl_ctx, const char *min_version)
     }
 #endif
 
-    const char *compiletime_min_version = "1.2";
+    const char *compiletime_min_version = "1.3";
```

Comment:
I would recommend enabling this as a transition across releases.
So  the default TLS in the first release is 1.2, but 1.3 is supported and encouraged.
In the next release, 1.3 is the default, as most CFEngine clients/servers support 1.3 at this point.

This relates to my comment below, as I would also not recommend to allow the client/server to simply pick a lower TLS version as its discretion.